be/src/olap/rowset/segment_v2/segment_writer.cpp

```diff
@@ -752,7 +752,12 @@ Status SegmentWriter::finalize_columns_data() {
     if (_has_key) {
         _row_count = _num_rows_written;
     } else {
-        CHECK_EQ(_row_count, _num_rows_written);
+        if (_row_count != _num_rows_written) {
```

Comment:
Add a DCHECK here, which can help us to find the issue quickly in test environment.

Reply:
> 

Thanks for comment, now i have added a DCHECK, take a look please